model/course.go

```diff
@@ -68,6 +69,54 @@ func (c *Course) SetCameraPresetPreference(pref []CameraPresetPreference) {
 	c.CameraPresetPreferences = string(pBytes)
 }
 
+type SourcePreference struct {
+	LectureHallID uint `json:"lecture_hall_id"`
+	SourceMode    int  `json:"source_mode"`
+}
+
+// GetSourcePreference retrieves the source preferences
+func (c Course) GetSourcePreference() []SourcePreference {
+	var res []SourcePreference
+	err := json.Unmarshal([]byte(c.SourcePreferences), &res)
+	if err != nil {
+		return []SourcePreference{}
+	}
+	return res
+}
+
+// GetSourceModeForLectureHall retrieves the source preference for the given lecture hall, returns default SourcePreference if non-existing
+func (c Course) GetSourceModeForLectureHall(id uint) int {
+	for _, preference := range c.GetSourcePreference() {
+		if preference.LectureHallID == id {
+			return preference.SourceMode
+		}
+	}
+	return 0
+}
+
+// CanUseSource returns whether the specified source type is allowed for the lecture hall id given
+func (c Course) CanUseSource(lectureHallID uint, sourceType string) bool {
+	mode := c.GetSourceModeForLectureHall(lectureHallID)
+	switch sourceType {
+	case "PRES":
+		return mode != 2
+	case "CAM":
+		return mode != 1
+	case "COMB":
+		return mode != 1 && mode != 2
+	}
+	return false
```

Comment:
In case something unexpected happens, this should default to true

Reply:
Done.